### PR TITLE
Updates to RequestView table and RecentDeliveries table

### PR DESCRIPTION
**Pull request by ao508**
General refactoring done:
- use of react table components
- organized content by pages

Added logos.

Added a navbar.

Misc:
- placeholder for home page content

Signed-off-by: Angelica Ochoa <15623749+ao508@users.noreply.github.com>

--

TODO:
- [x] switch to functional components (remove classes)
- [ ] also type functions as React.FunctionComponent (see comment below)
- [x] helper functions should be defined outside of components 

**Moved to separate cards:**
- add offset/limit to [graphql query](https://neo4j.com/docs/graphql-manual/current/pagination/offset-based/) instead of in react table 
- use moment.js for date formatting **--> made into separate card**
- use of `types` instead of classes for extending data models (might not need to do this if what we need is achieved by other means) // **NOTE: most of these were switched to `type` declarations but there are still a couple cases where this hasn't been done yet because of uncertainty on how to convert one data model `type` to another**  **--> made into separate card**

### CODE REVIEW

File: src/pages/recentDeliveries/RecentDeliveriesPage.tsx [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+  const { loading, error, data } = useQuery(ExampleQueryDocument);
+
+  function transformAndFilterRequestsByDate(requestsList: Array<Request>) {
+    const referenceDate = new Date();
+    referenceDate.setDate(referenceDate.getDate() - 60);
+    const refDateAsDate = new Date(referenceDate);
+    let month = refDateAsDate.getMonth() + 1;
+    let day = refDateAsDate.getDate();
+
+    const refDateFilter = `${refDateAsDate.getFullYear()}-${
```
Comment (alisman) [inline src/pages/recentDeliveries/RecentDeliveriesPage.tsx:43]: I would use Moment.js for date formatting.  Save yourself a lot of trouble.
https://github.com/moment/moment

File: src/pages/recentDeliveries/RecentDeliveriesTable.tsx [hunk 2]
```diff
@@ -0,0 +1,122 @@
...
+    sortable: true,
+    filterable: true
+  }
+];
+
+export interface RecentDeliveriesTableProps {
+  data: RequestWithDate[];
+}
+
+export default class RecentDeliveriesTable extends React.Component<
```
Comment (alisman) [inline src/pages/recentDeliveries/RecentDeliveriesTable.tsx:103]: always use react functional components.  in TS they can be defined like so:

export const MyComponent: React.FunctionComponent<MyComponentProps> = (
    props
) => {
     return <div>{..whatever}</div>
}

File: src/pages/requestView/RequestViewPage.tsx [hunk 3]
```diff
@@ -1,17 +1,78 @@
...
+      var currentSm = sample.hasMetadataSampleMetadata[i];
+      if (currentSm.importDate >= latestDeliv.importDate) {
+        latestDeliv = currentSm;
+      }
+    }
+    return [firstDeliv, latestDeliv];
+  }
+}
+
+export function RequestSummary() {
```
Comment (alisman) [inline src/pages/requestView/RequestViewPage.tsx:44]: See above this should be typed as React Function component

File: src/pages/recentDeliveries/RecentDeliveriesPage.tsx [hunk 4]
```diff
@@ -0,0 +1,71 @@
...
+    }
+    smDataList.sort();
+    return [smDataList[0], smDataList[smDataList.length - 1]];
+  }
+}
+
+function RecentDeliveriesPage() {
+  const { loading, error, data } = useQuery(ExampleQueryDocument);
+
+  function transformAndFilterRequestsByDate(requestsList: Array<Request>) {
```
Comment (alisman) [inline src/pages/recentDeliveries/RecentDeliveriesPage.tsx:36]: helper function should be defined outside of component and then memoized using useMemo hook

File: src/pages/recentDeliveries/RecentDeliveriesPage.tsx [hunk 5]
```diff
@@ -0,0 +1,71 @@
+import "./RecentDeliveries.css";
+import Container from "react-bootstrap/Container";
+import RecentDeliveriesTable from "./RecentDeliveriesTable";
+import { Request, ExampleQueryDocument } from "../../generated/graphql";
+import { useQuery } from "@apollo/client";
+
+export class RequestWithDate {
```
Comment (alisman) [inline src/pages/recentDeliveries/RecentDeliveriesPage.tsx:7]: Using classes in js is frowned upon. You should try to use functional approach.  We can discuss this.

File: src/pages/recentDeliveries/RecentDeliveriesPage.tsx (hunk 1 above)
Comment (ao508) [reply src/pages/recentDeliveries/RecentDeliveriesPage.tsx:43]: 🙏

File: src/pages/recentDeliveries/RecentDeliveriesPage.tsx (hunk 4 above)
Comment (ao508) [reply src/pages/recentDeliveries/RecentDeliveriesPage.tsx:36]: Adding this comment as a reminder to ask about this tomorrow

File: src/pages/recentDeliveries/RecentDeliveriesPage.tsx [hunk 6]
```diff
@@ -0,0 +1,71 @@
...
+  constructor(smileRequest: Request) {
+    this.smileRequest = smileRequest;
+    this.totalSamples = smileRequest.hasSampleSamples.length;
+    let deliveryDates = this.resolveAndSortDeliveryDates(smileRequest);
+    this.firstDelivered = deliveryDates[0];
+    this.lastDateUpdated = deliveryDates[1];
+  }
+  private resolveAndSortDeliveryDates(request: Request) {
+    var smDataList: string[] = [];
+    for (var i = 0; i < request.hasSampleSamples.length; i++) {
```
Comment (alisman) [inline src/pages/recentDeliveries/RecentDeliveriesPage.tsx:21]: basically we want to never use native looping.  use map, filter, reduce or lodash funcions

File: src/pages/home/HomePage.tsx [hunk 7]
```diff
@@ -0,0 +1,8 @@
+import * as React from "react";
+import Container from "react-bootstrap/Container";
+
+function HomePage() {
```
Comment (alisman) [inline src/pages/home/HomePage.tsx:4]: functional component.   type this.

**Pull request merged**